bubblewrap.c

```diff
@@ -792,6 +812,93 @@ privileged_op (int         privileged_op_socket,
     }
 }
 
+struct _StringBuilder
+{
+  char * str;
+  size_t size;
+  size_t offset;
+};
+
+static void
+strappend(struct _StringBuilder *dest, const char *src)
+{
+  size_t len = strlen(src);
+  if (dest->offset + len >= dest->size) {
+    dest->size = (dest->size + len) * 2;
+    dest->str = realloc(dest->str, dest->size);
+    if (dest->str == NULL)
+      die ("Out of memory");
+  }
+
+  strcpy(dest->str + dest->offset, src);
+  dest->offset += len;
+}
+
+/*
+ * "/hello:/goodbye" -> "lowerdir=/oldroot/hello:/oldroot/goodbye"
+ */
+static char *
+ro_overlay_options(const char* layers)
+{
+  struct _StringBuilder sb = {0};
+  cleanup_free char *layers_mut = strdup(layers);
+  char buf[PATH_MAX];
+  char * token;
+  int first = 1;
+
+  strappend(&sb, "lowerdir=");
+
+  token = strtok(layers_mut, ":");
```

Comment:
I'm not really happy about the amount of `strtok()` here: C string manipulation is notoriously easy to get wrong in security-sensitive ways, and bubblewrap is sometimes setuid root.

I'd prefer it if the layers can be defined in a way that doesn't need to be parsed, perhaps something like

```
--overlay-upper /foo/upper --overlay-lower /lower1 --overlay-lower /lower2 [--overlay-rw /foo/work] --overlay-onto /foo/merged
```

with these options only allowed in exactly that order (similar to how the `--perms` and `--size` modifiers work).